checks/linux.py

```diff
@@ -0,0 +1,172 @@
+                From a technical standpoint a Desktop Environment should not advertise the availability of these captures if they are not implemented. This behavior can therefore be considered a bug in your Desktop Environment."""]
+
+    return [LEVEL_INFO, "X11", ""]
+
+
+def checkDesktopEnvironment(lines):
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    desktopEnvironmentLine = search('Desktop Environment:', lines)
+    desktopEnvironment = desktopEnvironmentLine[0].split()
+
+    if (len(desktopEnvironment) > 3):
+        desktopEnvironment = desktopEnvironment[3:]
+        desktopEnvironment = ' '.join(desktopEnvironment)
+    else:
+        desktopEnvironment = ''
+
+    if (len(desktopEnvironment) > 0):
+        return [LEVEL_INFO, desktopEnvironment, '']
+
+
+def checkMissingModules(lines):
+    isDistroNix = search('Distribution:', lines)
+
+    if (len(isDistroNix) <= 0):
+        return
+
+    modulesMissingList = []
+    modulesCheckList = ('obs-browser.so', 'obs-websocket.so', 'vlc-video.so')
+
+    for module in modulesCheckList:
+        if not search(module, lines):
+            modulesMissingList.append(module)
+
+    if len(modulesMissingList):
+        modulesMissingString = str(modulesMissingList)
+        modulesMissingString = modulesMissingString.replace("['", "<ul><li>")
+        modulesMissingString = modulesMissingString.replace("', '", "</li><li>")
+        modulesMissingString = modulesMissingString.replace("']", "</li></ul>")
+
+        return [LEVEL_INFO, "Missing Modules (" + str(len(modulesMissingList)) + ")",
+                """You are missing the following default modules:<br>""" + modulesMissingString]
+
+
+def checkLinuxVCam(lines):
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    hasV4L2Module = search('v4l2loopback not installed', lines)
+
+    if len(hasV4L2Module) > 0:
+        return [LEVEL_INFO, "Virtual Camera not available",
+                """Using the Virtual Camera requires the <code>v4l2loopback</code> kernel module to be installed.<br>
+                If required, please refer to our <a href="https://github.com/obsproject/obs-studio/wiki/install-instructions#prerequisites-for-all-versions">Install Instructions</a> on how to install this on your distribution.<br>
+                If the module was not already loaded, OBS will normally ask you for permission to load it when required. This requires a working <code>polkit</code> setup.<br>
+                You can also load the module manually using <code>modprobe v4l2loopback exclusive_caps=1 card_label='OBS Virtual Camera'</code>."""]
```

Comment:
This feels a bit long-winded, and if this is a common issue, we should be updating the virtual camera troubleshooting page to include steps to resolve common Linux vcam issues and just linking that instead. Publicly indexable pages are far, far better than log analyzer messages.

Reply:
@tiberium-v would probably know better then me, but from what I can remember it happens primarily with Arch users because they never had reason to set up `polkit`. I think most OOTB-distros come with it preinstalled these days because some system settings panels require it (e.g. setting an SDDM theme in KDE's system settings, the various partition managers, etc.).

Not sure where a good place for it would be. Maybe we could also put it in the [Prerequisites section](https://github.com/obsproject/obs-studio/wiki/install-instructions#prerequisites-for-all-versions)? Since that's where users are pointed to anyway.

Reply:
We have a page for vcam stuff already, here: https://obsproject.com/kb/virtual-camera-troubleshooting so it should go there. I understand it's important to have the installation pre-requisite in the install instructions, but specific troubleshooting steps should live in the troubleshooting guide.

Feel free to just ping me in the #documentation channel on Discord with what should be added there, and I can take care of it ASAP.

Reply:
Removed and [added to install instructions](https://github.com/obsproject/obs-studio/wiki/Install-Instructions/_compare/70e4df611cd4a982a21e9884522d62f3072a948c...6dd86287ec3440d8f064ad669ad39cbbc6912a95).

Reply:
As a note here, it was/is Arch and also an old Debian version. In essential, it can happen on every distro, when the user switched from a traditional DE to a WM and forget to load polkit by them self.